bundle/edu.gemini.horizons.api/src/main/scala/edu/gemini/horizons/server/backend/HS2.scala

```diff
@@ -1,29 +1,55 @@
 package edu.gemini.horizons.server.backend
 
-import edu.gemini.spModel.core.HorizonsDesignation.{MajorBody, AsteroidOldStyle, AsteroidNewStyle, Comet}
-
-import java.util.logging.{Level, Logger}
-
-import edu.gemini.spModel.core._
-import org.apache.commons.httpclient.HttpClient
-import org.apache.commons.httpclient.HttpException
-import org.apache.commons.httpclient.NameValuePair
-import org.apache.commons.httpclient.methods.GetMethod
-import edu.gemini.horizons.api._
-import java.io.IOException
-import CgiHorizonsConstants._
+import java.io.{IOException, InputStream}
+import java.net.{URL, URLConnection, URLEncoder}
+import java.nio.charset.Charset
 import java.time.ZoneOffset.UTC
 import java.time.format.DateTimeFormatter
-import java.util.{Calendar, Date}
 import java.util.Locale.US
+import java.util.logging.{Level, Logger}
+import java.util.{Calendar, Date}
+import javax.net.ssl.{HostnameVerifier, HttpsURLConnection, SSLSession}
+
+import edu.gemini.horizons.api._
+import edu.gemini.horizons.server.backend.CgiHorizonsConstants._
+import edu.gemini.spModel.core.HorizonsDesignation.{AsteroidNewStyle, AsteroidOldStyle, Comet, MajorBody}
+import edu.gemini.spModel.core._
+import edu.gemini.util.ssl.GemSslSocketFactory
 
 import scala.collection.JavaConverters._
-import scala.util.matching.Regex
 import scala.io.Source
-import scalaz._, Scalaz._, scalaz.effect.{ IO, Resource }
+import scala.util.matching.Regex
+import scalaz.Scalaz._
+import scalaz._
+import scalaz.effect.{IO, Resource}
 
 // work-in-progress; this will get a better name and will probably move elsewhere
 object HorizonsService2 {
+  private val LOG = Logger.getLogger(this.getName)
+
+  // Lets any remote endpoint be connectable
+  private val hostnameVerifier: HostnameVerifier = new HostnameVerifier {
+     def verify(s: String, sslSession: SSLSession) = true
+  }
+
+  private val timeout = 3 * 60000  // 1 min ?
+
+  private object ConnectionCharset {
+    val default = Charset.forName("UTF-8")
+
+    def set(conn: URLConnection) {
+      conn.setRequestProperty("Accept-Charset", default.displayName())
+    }
+
+    def get(conn: URLConnection): Charset =
+      Option(conn.getContentType).getOrElse("").replace(" ", "").split(';').find {
```

Comment:
So `orZero` just relies on an underlying `Monoid` and retrieves the operation identity?
Sensible.